cl/citations/tests.py

```diff
@@ -1085,3 +1093,267 @@ def _print_failed_cases(
+                        id=4,
+                        score=0,
+                    ),
+                ],
+                [
+                    DummyParenthetical(
+                        text="Despite differing facts emphasized by the majority and dissent, the majority held that “Respondent’s crime was solely a reporting offense”",
+                        id=5,
+                        score=0,
+                    ),
+                ],
+                [
+                    DummyParenthetical(
+                        text="“[D]espite the differences between restitution and a traditional fine, restitution still implicates the prosecutorial powers of government[.]”",
+                        id=6,
+                        score=0,
+                    ),
+                ],
+            ),
+            (
+                [
+                    DummyParenthetical(
+                        text="Finding valid a federal law criminalizing the destruction or mutilation of a draft registration against a First Amendment challenge",
+                        id=10,
+                        score=0,
+                    ),
+                ],
+                [
+                    DummyParenthetical(
+                        text="Applying medium scrutiny test to state action having an incidental effect on right to free expression ",
+                        id=11,
+                        score=0,
+                    )
+                ],
+            ),
+            (
+                [
+                    DummyParenthetical(
+                        text="The loss of First Amendment freedoms, for even minimal period of time, unquestionably constitutes irreparable injury.",
+                        id=11,
+                        score=0,
+                    ),
+                    DummyParenthetical(
+                        text="The loss of First Amendment freedoms, for even minimal period of time, unquestionably constitutes irreparable injury.",
+                        id=12,
+                        score=0,
+                    ),
+                    DummyParenthetical(
+                        text="The loss of First Amendment freedoms, for even minimal period of time, unquestionably constitutes irreparable injury.",
+                        id=13,
+                        score=0,
+                    ),
+                ],
+                [
+                    DummyParenthetical(
+                        text=" Holding public employees could not be fired because of their politics unless they held “policymaking” or “confidential” positions ",
+                        id=14,
+                        score=0,
+                    ),
+                ],
+            ),
+        ]
+        for i, groups in enumerate(expected_groups):
+            with self.subTest(f"Testing {groups} are grouped correctly.", i=i):
+                flat = sum(groups, start=[])
+                output_groups = get_parenthetical_groups(flat)
```

Comment:
This is a clever approach, but it took me a while to figure out what you're doing. The test cases have things grouped properly and so you ungroup them by flattening them, then you regroup them using your code. Let's get a comment or two explaining this for me the next time I read this code, so I don't have to figure it out again.

Reply:
Will do!

Reply:
Agree, this might be a bit too clever for its own good. Maybe a bit more readable would be:

```suggestion
                flat = list(chain.from_iterable(groups))
                output_groups = get_parenthetical_groups(flat)
```

(`from itertools import chain`)